_stbt/core.py

```diff
@@ -1622,10 +1637,14 @@ def teardown(self):
             source = None
         if not self.novideo:
```

Comment:
Hmm.  We are only setting state to NULL if we haven't seen `NoVideo`.  This doesn't seem appropriate.